### PR TITLE
Purchases number list

### PR DESCRIPTION
**Pull request by R-Bear**
# Purchase number list Design
### routes
- [x] admin/purchases/show
 ` namespace :admin do`
 ` get 'purchases/show'`
`end`
### views
- [x] admin/purchases/show.html.erb
### assets
 - [x] admin/purchases.scss
 - [x] admin/purchases_chart.js

### CODE REVIEW

File: pj_movie/app/assets/javascripts/admin/purchase_chart.js [hunk 1]
```diff
@@ -0,0 +1,968 @@
+window.onload = function () {
```
Comment (N310) [inline pj_movie/app/assets/javascripts/admin/purchase_chart.js]: do you use `canvasjs` ?

File: pj_movie/app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -1,3 +1 @@
-// Place all the styles related to the StaticPages controller here.
-// They will automatically be included in application.css.
-// You can use Sass (SCSS) here: http://sass-lang.com/
+
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/static_pages.scss]: Do you need this change in the branch ?

File: pj_movie/spec/controllers/admin/purchases_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe Admin::PurchasesController, type: :controller do
```
Comment (N310) [inline pj_movie/spec/controllers/admin/purchases_controller_spec.rb]: we should use request spec for controller test.

- Reference
https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec

File: pj_movie/config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,8 @@
 Rails.application.routes.draw do
 
+  namespace :admin do
+  get 'purchases/show'
```
Comment (N310) [inline pj_movie/config/routes.rb]: you had better add indent, use `resources` and change to `index` action.

File: pj_movie/app/views/admin/purchases/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<div class="purchaselist_container">
+<div id="chartContainer" "></div>
```
Comment (N310) [inline pj_movie/app/views/admin/purchases/show.html.erb]: plz add indent.
you should change horizontal axis from date to movies.

File: pj_movie/app/views/admin/purchases/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  <br>
+  <div class="table_container">
+    <h1>Table of Purchase for November</h1>
+    <table >
```
Comment (N310) [inline pj_movie/app/views/admin/purchases/show.html.erb]: I recommend that you remove unneeded space.
`<table >` → `<table>`

File: pj_movie/config/routes.rb [hunk 7]
```diff
@@ -1,5 +1,8 @@
...
+  end
```
Comment (N310) [inline pj_movie/config/routes.rb]: You move to this block under the root.

File: pj_movie/app/assets/javascripts/admin/purchase_chart.js (hunk 1 above)
Comment (R-Bear) [reply pj_movie/app/assets/javascripts/admin/purchase_chart.js]: Yes, I do.

File: pj_movie/app/assets/stylesheets/static_pages.scss (hunk 2 above)
Comment (R-Bear) [reply pj_movie/app/assets/stylesheets/static_pages.scss]: No, I do not change static_branch

File: pj_movie/config/routes.rb [hunk 8]
```diff
@@ -3,4 +3,8 @@
   root 'static_pages#home'
   get 'static_pages/home'
   get 'static_pages/help'
```
Comment (PechBunsinh) [inline pj_movie/config/routes.rb]: you should remove this line if you don't need

File: pj_movie/app/assets/stylesheets/admin/purchases.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
+.detail_content, .purchaselist_container{
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/admin/purchases.scss]: you should add a space after css selector.
ex. `.detail_content, .purchaselist_container {`

File: pj_movie/app/assets/stylesheets/adnmin/purchases.scss [hunk 10]
```diff
@@ -0,0 +1,47 @@
+.purchaselist_container{
```
Comment (N310) [inline pj_movie/app/assets/stylesheets/adnmin/purchases.scss]: Do you need this file?

File: pj_movie/app/controllers/admin/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+class Admin::UsersController < ActionController::Base
+  def new
+  end
+  def login
+  end
+end
```
Comment (PechBunsinh) [inline pj_movie/app/controllers/admin/users_controller.rb]: you should remove this file if you don't need

File: pj_movie/app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -1,9 +1,9 @@
 <!DOCTYPE html>
 <html>
   <head>
-  <title><%= yield(:title) %> | Project Movie</title>
-  <%= render 'layouts/rails_default' %>
-  <%= render 'layouts/shim' %>
+    <title><%= yield(:title) %> | Project Movie</title>
```
Comment (loemhuolay) [inline pj_movie/app/views/layouts/application.html.erb]: also remove this "|"

File: pj_movie/Gemfile [hunk 13]
```diff
@@ -10,6 +10,13 @@ end
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 5.1.6'
 
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3'
+
```
Comment (loemhuolay) [inline pj_movie/Gemfile]: you should delete these line, we no need to use SQLite3

File: pj_movie/Gemfile.lock [hunk 14]
```diff
@@ -170,6 +177,7 @@ GEM
       actionpack (>= 4.0)
       activesupport (>= 4.0)
       sprockets (>= 3.0.0)
+    sqlite3 (1.4.1)
```
Comment (loemhuolay) [inline pj_movie/Gemfile.lock]: should delete sqlite3

File: pj_movie/app/assets/stylesheets/admin/password_resets.scss [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+      align-items: center;
+      justify-content: center;
+      padding: 0;
+      height: 100vh;
+      .forgot_password {
+        display: flex;
+        flex-direction: column;
+        justify-content: left;
+        font-size: 16px;
+        font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/password_resets.scss]: I think no need font Roboto

File: pj_movie/app/assets/stylesheets/admin/password_resets.scss [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+          display: flex;
+          flex-direction: column;
+          justify-content: center;
+          align-items: center;
+          text-align: left;
+          font-size: 16px;
+          border: none;
+          padding: 0;
+          width: 95%;
+          font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/password_resets.scss]: font Roboto

File: pj_movie/app/assets/stylesheets/admin/top_page.scss [hunk 17]
```diff
@@ -166,14 +162,16 @@
   }
 }
 //===================================Container=========================
-.container-admin {
+.container.container-admin {
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/top_page.scss]: I think use "_" better than "-"
Ex: container-admin => container_admin

File: pj_movie/app/assets/stylesheets/admin/users.scss [hunk 18]
```diff
@@ -0,0 +1,11 @@
+.admin_signin {
+  .signin_container{
+    margin-top: 98px;
+    background-color: #fafafa;
+  }
+  .signup_container{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/admin/users.scss]: Add space before "{"

File: pj_movie/app/assets/stylesheets/custom.scss [hunk 19]
```diff
@@ -83,8 +83,8 @@ textarea {
           }
         }
       }
-      &:hover .dropdown_content {
-        display: block;
+      &:hover .dropdown_content{
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/custom.scss]: you should add space 
Ex:  &:hover .dropdown_content{ =>  &:hover .dropdown_content {

File: pj_movie/app/assets/stylesheets/password_resets.scss [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      padding: 0;
+      .forgot_password {
+        display: flex;
+        flex-direction: column;
+        justify-content: left;
+        font-size: 16px;
+        font-family: Roboto;
```
Comment (loemhuolay) [inline pj_movie/app/assets/stylesheets/password_resets.scss]: Roboto

File: pj_movie/app/assets/stylesheets/admin/top_page.scss (hunk 17 above)
Comment (R-Bear) [reply pj_movie/app/assets/stylesheets/admin/top_page.scss]: your comment about container-admin is the structure on top page, can I change structure on top page?

**Pull request merged**